core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -51,9 +56,59 @@ public String getContents() {
             return contents;
         }
     }
-    
-    public static NamedStringTag newYamlSpecTag(String contents) { return new NamedStringTag(YAML_SPEC_KIND, contents); }
-    public static NamedStringTag newNotesTag(String contents) { return new NamedStringTag(NOTES_KIND, contents); }
+
+    public static class ListTag<T> {
```

Comment:
I think we shouldn't reuse this, if there are multiple items each should be inserted in a separate tag (i.e. have flat tags). How does adding a trait look like with this type of tag? Should it be inserted in a new collection of traits? Just thinking out loud.